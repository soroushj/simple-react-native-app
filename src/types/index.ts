export interface IRootState {
  counter: number;
}
