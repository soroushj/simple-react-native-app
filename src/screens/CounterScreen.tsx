import React from 'react';
import { View, Text, Button } from 'react-native';
import { NavigationScreenOptions } from 'react-navigation';
import { connect } from 'react-redux';
import { ICounterScreenProps } from './CounterScreen.types';
import { IRootState } from '../types';
import { counterIncrement } from '../actions/counter';
import styles from './styles';

class CounterScreen extends React.Component<ICounterScreenProps> {
  static navigationOptions: NavigationScreenOptions = {
    title: 'Counter',
  };

  onIncrementPress = () => {
    this.props.counterIncrement();
  };

  onBackPress = () => {
    this.props.navigation.navigate('Home');
  };

  render() {
    return (
      <View style={styles.screenContainer}>
        <View style={styles.elementContainer}>
          <Text>Counter value: {this.props.counterValue}</Text>
        </View>
        <View style={styles.elementContainer}>
          <Button title=" +1 " onPress={this.onIncrementPress} />
        </View>
        <View style={styles.elementContainer}>
          <Button title=" Back " onPress={this.onBackPress} />
        </View>
      </View>
    );
  }
}

const mapStateToProps = (state: IRootState) => {
  return { counterValue: state.counter };
};

const mapDispatchToProps = { counterIncrement };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CounterScreen);
