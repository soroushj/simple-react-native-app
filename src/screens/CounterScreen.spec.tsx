import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CounterScreen } from './CounterScreen';
import { ICounterScreenProps } from './CounterScreen.types';

const setup = () => {
  const props: ICounterScreenProps = {
    navigation: { navigate: jest.fn() } as any,
    counterValue: 10,
    counterIncrement: jest.fn(),
  };
  const wrapper = shallow(<CounterScreen {...props} />);
  return { wrapper, props };
};

test('CounterScreen snapshot', () => {
  const { wrapper } = setup();
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('CounterScreen, incrementButton press', () => {
  const { wrapper, props } = setup();
  const incrementButton = wrapper.find('[testID="incrementButton"]');
  const incrementButtonProps = incrementButton.props() as any;
  incrementButtonProps.onPress();
  expect(props.counterIncrement).toBeCalled();
});

test('CounterScreen, backButton press', () => {
  const { wrapper, props } = setup();
  const backButton = wrapper.find('[testID="backButton"]');
  const backButtonProps = backButton.props() as any;
  backButtonProps.onPress();
  expect(props.navigation.navigate).toBeCalledWith('Home');
});
