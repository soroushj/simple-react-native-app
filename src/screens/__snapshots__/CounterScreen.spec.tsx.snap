// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterScreen snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Component>
      Counter value: 
      10
    </Component>
  </Component>
  <Component
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Button
      onPress={[Function]}
      testID="incrementButton"
      title=" +1 "
    />
  </Component>
  <Component
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Button
      onPress={[Function]}
      testID="backButton"
      title=" Back "
    />
  </Component>
</Component>
`;
